ragna/_ui/api_wrapper.py

```diff
@@ -1,70 +1,70 @@
-
-        # FIXME: the token should come from a cookie that is set after UI login
-        from ragna.core._rag import default_user  # noqa
-
-        user = default_user()
-        token = (
-            self.client.post(
-                "/token",
-                data={
-                    "username": user,
-                    "password": os.environ.get(
-                        "AI_PROXY_DEMO_AUTHENTICATION_PASSWORD", user
-                    ),
-                },
-            )
-            .raise_for_status()
-            .json()
-        )
-        # FIXME: remove this as it should come from a cookie on the JS side as well
-        self.token = token
-        self.client.headers["Authorization"] = f"Bearer {token}"
+class RagnaAuthTokenExpiredException(Exception):
+    """Just a wrapper around Exception"""
 
-    async def auth(self, username, password):
-        async with httpx.AsyncClient(
-            base_url=self.client.base_url, headers=self.client.headers
-        ) as async_client:
-            self.token = (
-                (
-                    await async_client.post(
-                        "/token",
-                        data={"username": username, "password": password},
-                    )
-                )
-                .raise_for_status()
-                .json()
-            )
+    pass
 
-            self.client.headers["Authorization"] = f"Bearer {self.token}"
 
-            return True
+# The goal is this class is to provide ready-to-use functions to interact with the API
+class ApiWrapper(param.Parameterized):
+    auth_token = param.String(default=None)
 
-    def get_chats(self):
-        json_data = self.client.get("/chats").raise_for_status().json()
+    def __init__(self, api_url, **params):
+        self.client = httpx.AsyncClient(base_url=api_url)
 
-        for chat in json_data:
-            chat["messages"] = [self.improve_message(msg) for msg in chat["messages"]]
+        httpx.get(api_url).raise_for_status()
 
-        return json_data
+        super().__init__(**params)
 
-    def answer(self, chat_id, prompt):
+    async def auth(self, username, password):
+        self.auth_token = (
+            (
+                await self.client.post(
+                    "/token",
+                    data={"username": username, "password": password},
+                )
+            )
+            .raise_for_status()
+            .json()
+        )
+
+        return True
+
+    @param.depends("auth_token", watch=True)
+    def update_auth_header(self):
+        self.client.headers["Authorization"] = f"Bearer {self.auth_token}"
+
+    async def get_chats(self):
+        try:
+            json_data = (await self.client.get("/chats")).raise_for_status().json()
+            for chat in json_data:
+                chat["messages"] = [
+                    self.improve_message(msg) for msg in chat["messages"]
+                ]
+            return json_data
+
+        except httpx.HTTPStatusError as e:
```

Comment:
Didn't we want to do that in `__init__` and as part of getting the components?

Reply:
Done in my last commit